---
title: "WP2 Overview"
output: html_document
date: "2024-11-26"
---
## Libraries & Setup
```{r setup}
library(tidyverse)
library(reshape2)
library(ggridges)
library(vegan)
library(readr)
library(breakaway)
library(rioja)

# Here we have made our input into smaller 100MB files and now read them in
setwd("rawdata/MetagenomicFirstRound/")
con <- pipe("cat part.gz_* | gunzip -c", "rb")
data <- read_csv(con)
close(con)
setwd("../../")
#metadata
metadata <- read.csv("metadata.csv")

# Minimum amount of damage filter
DamMin = 0.0
D_Max = 1.0
#Lambda Likelihood Ratio
LR = 0
# Minimum reads for parsing taxa
MinRead = 1
# Minimum mean read length
MinLength = 35
```

## Metadata and seq mapping observations

Here we are looking at some values produced by fastp.

First lets look at the relationship between duplication percentage and ∆Ct (difference between the ideal number fo PCR cycles and used number of cycles). The est age of samples in BP is shown. 

```{r}
plot(metadata$Dct,metadata$duplication.rate,pch=16,ylab="% duplication")
abline(v=0,col="darkred",lwd=3)
text(0.2,60,labels="Underamplified", adj=0)
text(-0.2,60,labels="Overamplified", adj=1)
text(metadata$Dct,metadata$duplication.rate+1.5,labels = metadata$age,cex=0.4)
```

Now lets look at the low complexity proportion vs ∆Ct

```{r }
plot(metadata$Dct,metadata$low.complexity,pch=16,ylab="% low complexity")
abline(v=0,col="darkred",lwd=3)
text(0.2,0.32,labels="Underamplified", adj=0)
text(-0.2,0.32,labels="Overamplified", adj=1)
```

Now how many post-fastp reads there are per sample.

```{r }
plot(metadata$Dct,metadata$dedupReads,pch=16,ylab="Post fastp reads")
abline(v=0,col="darkred",lwd=3)
text(0.2,2.2*10^8,labels="Underamplified", adj=0)
text(-0.2,2.2*10^8,labels="Overamplified", adj=1)
text(metadata$Dct,metadata$dedupReads+5*10^6,labels = metadata$age,cex=0.4)

```

## Age-Damage / Length relationships 

First we wrangle the raw data

```{r }
# Let's make an animal subset of the data 
df1 <- data %>% filter(MAP_damage > DamMin, N_reads >= MinRead, mean_L > MinLength, MAP_significance  > LR,  grepl("Metazoa",tax_path), grepl("^genus", tax_rank), grepl("", sample))
inputdata <- df1
inputdata$sample2 <- factor(inputdata$sample,levels= sort(unique(inputdata$sample),decreasing = TRUE))
inputdata$age <- metadata$age[match(inputdata$sample2,metadata$Sample)]
inputdata$site <- as.factor(metadata$CoreID[match(inputdata$sample2,metadata$Sample)])

raw.metazoa <- inputdata[inputdata$N_reads>99,]
raw.metazoa.50 <- inputdata[inputdata$N_reads>49,]
raw.metazoa.1  <- inputdata

raw.metazoa.50.wide <- raw.metazoa.50 %>%
  pivot_wider(
    id_cols = tax_name,         # Rows will be `tax_name`
    names_from = sample,        # Columns will be `sample`
    values_from = N_reads,      # Observations will be `N_reads`
    values_fill = 0             # Fill missing values with 0
  )


raw.metazoa.1.wide <- raw.metazoa.1 %>%
  pivot_wider(
    id_cols = tax_name,         # Rows will be `tax_name`
    names_from = sample,        # Columns will be `sample`
    values_from = N_reads,      # Observations will be `N_reads`
    values_fill = 0             # Fill missing values with 0
  )


raw.metazoa.wide <- raw.metazoa %>%
  pivot_wider(
    id_cols = tax_name,         # Rows will be `tax_name`
    names_from = sample,        # Columns will be `sample`
    values_from = N_reads,      # Observations will be `N_reads`
    values_fill = 0             # Fill missing values with 0
  )



# plants
df1 <- data %>% filter(MAP_damage > DamMin, N_reads >= MinRead, mean_L > MinLength, MAP_significance  > LR,  grepl("Viridiplantae",tax_path), grepl("^genus", tax_rank), grepl("", sample))
inputdata <- df1
inputdata$sample2 <- factor(inputdata$sample,levels= sort(unique(inputdata$sample),decreasing = TRUE))
inputdata$age <- metadata$age[match(inputdata$sample2,metadata$Sample)]
inputdata$site <- as.factor(metadata$CoreID[match(inputdata$sample2,metadata$Sample)])

raw.plants <- inputdata[inputdata$N_reads>99,]
raw.plants.50 <- inputdata[inputdata$N_reads>49,]

raw.plants.50.wide <- raw.plants.50 %>%
  pivot_wider(
    id_cols = tax_name,         # Rows will be `tax_name`
    names_from = sample,        # Columns will be `sample`
    values_from = N_reads,      # Observations will be `N_reads`
    values_fill = 0             # Fill missing values with 0
  )


raw.plants.wide <- raw.plants %>%
  pivot_wider(
    id_cols = tax_name,         # Rows will be `tax_name`
    names_from = sample,        # Columns will be `sample`
    values_from = N_reads,      # Observations will be `N_reads`
    values_fill = 0             # Fill missing values with 0
  )


# bacteria 

df1 <- data %>% filter(MAP_damage > DamMin, N_reads >= MinRead, mean_L > MinLength, MAP_significance  > LR,  grepl("Bacteria",tax_path), grepl("^genus", tax_rank), grepl("", sample))
inputdata <- df1
inputdata$sample2 <- factor(inputdata$sample,levels= sort(unique(inputdata$sample),decreasing = TRUE))
inputdata$age <- metadata$age[match(inputdata$sample2,metadata$Sample)]
inputdata$site <- as.factor(metadata$CoreID[match(inputdata$sample2,metadata$Sample)])

raw.bacteria <- inputdata[inputdata$N_reads>99,]
raw.bacteria.50<- inputdata[inputdata$N_reads>49,]

raw.bacteria.50.wide <- raw.bacteria.50 %>%
  pivot_wider(
    id_cols = tax_name,         # Rows will be `tax_name`
    names_from = sample,        # Columns will be `sample`
    values_from = N_reads,      # Observations will be `N_reads`
    values_fill = 0             # Fill missing values with 0
  )


raw.bacteria.wide <- raw.bacteria %>%
  pivot_wider(
    id_cols = tax_name,         # Rows will be `tax_name`
    names_from = sample,        # Columns will be `sample`
    values_from = N_reads,      # Observations will be `N_reads`
    values_fill = 0             # Fill missing values with 0
  )



```

Now we make some plots, first the animals

```{r, fig.width=12,fig.height=9}
ggplot(raw.metazoa, aes(x = MAP_damage, y = sample2, group = sample2)) +
  geom_density_ridges2(scale = 3, alpha = 0.55) +  # Plot density ridges first
  geom_point(position = position_jitter(height = 0.1, width = 0), alpha = 0.6) + # Add points above ridges
  facet_wrap(~ site, scales = "free_y") + # Split into facets by coreID
  labs(y = "Samples (ordered by age)", x = "MAP Damage",title = "Metazoa MAP Damage")+
  geom_text(aes(x = max(MAP_damage), label = age), hjust = -0.1, size = 3,nudge_y = 0.4)

ggplot(raw.metazoa, aes(x = mean_L, y = sample2, group = sample2)) +
  geom_density_ridges2(scale = 3, alpha = 0.55) +  # Plot density ridges first
  geom_point(position = position_jitter(height = 0.1, width = 0), alpha = 0.6) + # Add points above ridges
  facet_wrap(~ site, scales = "free_y") + # Split into facets by coreID
  labs(y = "Samples (ordered by age)", x = "MAP Damage",title = "Metazoa Mean Length")+
  geom_text(aes(x = max(MAP_damage), label = age), hjust = -0.1, size = 3,nudge_y = 0.4)
```

Now bacteria

```{r , fig.width=12,fig.height=9}
ggplot(raw.bacteria, aes(x = MAP_damage, y = sample2, group = sample2)) +
  geom_density_ridges2(scale = 3, alpha = 0.55) +  # Plot density ridges first
  geom_point(position = position_jitter(height = 0.1, width = 0), alpha = 0.6) + # Add points above ridges
  facet_wrap(~ site, scales = "free_y") + # Split into facets by coreID
  labs(y = "Samples (ordered by age)", x = "MAP Damage",title = "Bacteria MAP Damage")+
  geom_text(aes(x = max(MAP_damage), label = age), hjust = -0.1, size = 3,nudge_y = 0.4)

ggplot(raw.bacteria, aes(x = mean_L, y = sample2, group = sample2)) +
  geom_density_ridges2(scale = 3, alpha = 0.55) +  # Plot density ridges first
  geom_point(position = position_jitter(height = 0.1, width = 0), alpha = 0.6) + # Add points above ridges
  facet_wrap(~ site, scales = "free_y") + # Split into facets by coreID
  labs(y = "Samples (ordered by age)", x = "MAP Damage",title = "Bacteria Mean Length")+
  geom_text(aes(x = max(MAP_damage), label = age), hjust = -0.1, size = 3,nudge_y = 0.4)
```


Now we make some plots, first the animals

```{r , fig.width=12,fig.height=9}
ggplot(raw.plants, aes(x = MAP_damage, y = sample2, group = sample2)) +
  geom_density_ridges2(scale = 3, alpha = 0.55) +  # Plot density ridges first
  geom_point(position = position_jitter(height = 0.1, width = 0), alpha = 0.6) + # Add points above ridges
  facet_wrap(~ site, scales = "free_y") + # Split into facets by coreID
  labs(y = "Samples (ordered by age)", x = "MAP Damage",title = "Plants MAP Damage")+
  geom_text(aes(x = max(MAP_damage), label = age), hjust = -0.1, size = 3,nudge_y = 0.4)

ggplot(raw.plants, aes(x = mean_L, y = sample2, group = sample2)) +
  geom_density_ridges2(scale = 3, alpha = 0.55) +  # Plot density ridges first
  geom_point(position = position_jitter(height = 0.1, width = 0), alpha = 0.6) + # Add points above ridges
  facet_wrap(~ site, scales = "free_y") + # Split into facets by coreID
  labs(y = "Samples (ordered by age)", x = "MAP Damage",title = "Plants Mean Length")+
  geom_text(aes(x = max(MAP_damage), label = age), hjust = -0.1, size = 3,nudge_y = 0.4)
```


## Richness Estimates 

Now let's look at some richness estimates per core

```{r}

richness <- raw.metazoa.50.wide %>%
  select(-tax_name) %>%
  summarise(across(everything(), ~ sum(. > 0))) %>%
  pivot_longer(cols = everything(), names_to = "Sample", values_to = "Richness")

# Combine richness with metadata
richness_plot_data <- richness %>%
  left_join(metadata, by = "Sample")


ggplot(richness_plot_data, aes(x = age, y = Richness, color = CoreID)) +
  geom_point(size = 4) +
  theme_minimal() +
  labs(
    title = "Metazoan Genus Richness vs. Age by Core (50 reads)",
    x = "Age",
    y = "Richness",
    color = "Core ID"
  ) +
  theme(
    plot.title = element_text(size = 16, hjust = 0.5),
    axis.title.x = element_text(size = 14),
    axis.title.y = element_text(size = 14),
    legend.title = element_text(size = 12)
  )

## More reads in the earlier core portion - lets use breakaway estimates 

results <- breakaway_nof1(raw.metazoa.1.wide[,-1])
named_estimates <- map_dbl(results, ~ .x$estimate)
names(named_estimates) <- names(results)  

diversity_df <- tibble(
  Sample = names(named_estimates),
  Richness = as.numeric(named_estimates)
)

# Merge with metadata
plot_data <- diversity_df %>%
  left_join(metadata, by = "Sample")

# Create the plot
ggplot(plot_data, aes(x = age, y = Richness, color = CoreID)) +
  geom_point(size = 4) +
  theme_minimal() +
  labs(
    title = "Metazoan Genus Richness (breakaway estimate) vs. Age by Core (50 reads)",
    x = "Age",
    y = "Richness",
    color = "Core ID"
  ) +
  theme(
    plot.title = element_text(size = 16, hjust = 0.5),
    axis.title.x = element_text(size = 14),
    axis.title.y = element_text(size = 14),
    legend.title = element_text(size = 12)
  )

# Analyze and plot for raw.plants.50.wide
richness_plants <- raw.plants.50.wide %>%
  select(-tax_name) %>%
  summarise(across(everything(), ~ sum(. > 0))) %>%
  pivot_longer(cols = everything(), names_to = "Sample", values_to = "Richness")

richness_plot_data_plants <- richness_plants %>%
  left_join(metadata, by = "Sample")

ggplot(richness_plot_data_plants, aes(x = age, y = Richness, color = CoreID)) +
  geom_point(size = 4) +
  theme_minimal() +
  labs(
    title = "Plant Genus Richness vs. Age by Core (50 reads)",
    x = "Age",
    y = "Richness",
    color = "Core ID"
  ) +
  theme(
    plot.title = element_text(size = 16, hjust = 0.5),
    axis.title.x = element_text(size = 14),
    axis.title.y = element_text(size = 14),
    legend.title = element_text(size = 12)
  )

# Analyze and plot for raw.bacteria.50.wide
richness_bacteria <- raw.bacteria.50.wide %>%
  select(-tax_name) %>%
  summarise(across(everything(), ~ sum(. > 0))) %>%
  pivot_longer(cols = everything(), names_to = "Sample", values_to = "Richness")

richness_plot_data_bacteria <- richness_bacteria %>%
  left_join(metadata, by = "Sample")

ggplot(richness_plot_data_bacteria, aes(x = age, y = Richness, color = CoreID)) +
  geom_point(size = 4) +
  theme_minimal() +
  labs(
    title = "Bacterial Genus Richness vs. Age by Core (50 reads)",
    x = "Age",
    y = "Richness",
    color = "Core ID"
  ) +
  theme(
    plot.title = element_text(size = 16, hjust = 0.5),
    axis.title.x = element_text(size = 14),
    axis.title.y = element_text(size = 14),
    legend.title = element_text(size = 12)
  )


```




## Beta Div

Now let's examine the structure between samples 

First the animals 

```{r, fig.width=14,fig.height=7.5}
raw.metazoa.50.wide.d <- as.data.frame(raw.metazoa.50.wide)
rownames(raw.metazoa.50.wide.d) <- raw.metazoa.50.wide.d[,1]
raw.metazoa.50.wide.d <- raw.metazoa.50.wide.d[,-c(1,2,3)]

par(mfrow = c(1, 2)) 
metaz.ord <- metaMDS(vegdist(t(raw.metazoa.50.wide.d),method = "jaccard",binary = TRUE),trymax = 200)
metaz.ord.b <- metaMDS(vegdist(t(raw.metazoa.50.wide.d),method = "bray"),trymax = 200)
plot(metaz.ord$points[,1],metaz.ord$points[,2],pch=16,col=as.numeric(as.factor(metadata$CoreID[match(rownames(metaz.ord$points),metadata$Sample)])),main="Jaccard 50")
#ordisurf(metaz.ord,metadata$age[match(rownames(metaz.ord$points),metadata$Sample)],add=TRUE)
text(metaz.ord$points[,1],metaz.ord$points[,2]+0.05,labels=metadata$age[match(rownames(metaz.ord$points),metadata$Sample)],cex=0.8)
legend("topleft",bty='n', col=c(1,2,3),pch=16,legend=levels(as.factor(metadata$CoreID[match(rownames(metaz.ord.b$points),metadata$Sample)])))

plot(metaz.ord.b$points[,1],metaz.ord.b$points[,2],pch=16,col=as.numeric(as.factor(metadata$CoreID[match(rownames(metaz.ord.b$points),metadata$Sample)])),main="Bray 50")
#ordisurf(metaz.ord,metadata$age[match(rownames(metaz.ord$points),metadata$Sample)],add=TRUE)
text(metaz.ord.b$points[,1],metaz.ord.b$points[,2]+0.05,labels=metadata$age[match(rownames(metaz.ord.b$points),metadata$Sample)],cex=0.8)
legend("topleft",bty='n', col=c(1,2,3),pch=16,legend=levels(as.factor(metadata$CoreID[match(rownames(metaz.ord.b$points),metadata$Sample)])))
```

Now plants...

```{r, fig.width=14,fig.height=7.5}
raw.plants.50.wide.d <- as.data.frame(raw.plants.50.wide)
rownames(raw.plants.50.wide.d) <- raw.plants.50.wide.d[,1]
raw.plants.50.wide.d <- raw.plants.50.wide.d[,-c(1,2,3)]

par(mfrow = c(1, 2)) 
metaz.ord <- metaMDS(vegdist(t(raw.plants.50.wide.d),method = "jaccard",binary = TRUE),trymax = 200)
metaz.ord.b <- metaMDS(vegdist(t(raw.plants.50.wide.d),method = "bray"),trymax = 200)
plot(metaz.ord$points[,1],metaz.ord$points[,2],pch=16,col=as.numeric(as.factor(metadata$CoreID[match(rownames(metaz.ord$points),metadata$Sample)])),main="Jaccard 50")
#ordisurf(metaz.ord,metadata$age[match(rownames(metaz.ord$points),metadata$Sample)],add=TRUE)
text(metaz.ord$points[,1],metaz.ord$points[,2]+0.05,labels=metadata$age[match(rownames(metaz.ord$points),metadata$Sample)],cex=0.8)
legend("topleft",bty='n', col=c(1,2,3),pch=16,legend=levels(as.factor(metadata$CoreID[match(rownames(metaz.ord.b$points),metadata$Sample)])))

plot(metaz.ord.b$points[,1],metaz.ord.b$points[,2],pch=16,col=as.numeric(as.factor(metadata$CoreID[match(rownames(metaz.ord.b$points),metadata$Sample)])),main="Bray 50")
#ordisurf(metaz.ord,metadata$age[match(rownames(metaz.ord$points),metadata$Sample)],add=TRUE)
text(metaz.ord.b$points[,1],metaz.ord.b$points[,2]+0.05,labels=metadata$age[match(rownames(metaz.ord.b$points),metadata$Sample)],cex=0.8)
legend("topleft",bty='n', col=c(1,2,3),pch=16,legend=levels(as.factor(metadata$CoreID[match(rownames(metaz.ord.b$points),metadata$Sample)])))
```


Finally the bacteria

```{r, fig.width=14,fig.height=7.5}
raw.bacteria.50.wide.d <- as.data.frame(raw.bacteria.50.wide)
rownames(raw.bacteria.50.wide.d) <- raw.bacteria.50.wide.d[,1]
raw.bacteria.50.wide.d <- raw.bacteria.50.wide.d[,-c(1,2,3)]

par(mfrow = c(1, 2)) 
metaz.ord <- metaMDS(vegdist(t(raw.bacteria.50.wide.d),method = "jaccard",binary = TRUE),trymax = 200)
metaz.ord.b <- metaMDS(vegdist(t(raw.plants.50.wide.d),method = "bray"),trymax = 200)
plot(metaz.ord$points[,1],metaz.ord$points[,2],pch=16,col=as.numeric(as.factor(metadata$CoreID[match(rownames(metaz.ord$points),metadata$Sample)])),main="Jaccard 50")
#ordisurf(metaz.ord,metadata$age[match(rownames(metaz.ord$points),metadata$Sample)],add=TRUE)
text(metaz.ord$points[,1],metaz.ord$points[,2]+0.05,labels=metadata$age[match(rownames(metaz.ord$points),metadata$Sample)],cex=0.8)
legend("topleft",bty='n', col=c(1,2,3),pch=16,legend=levels(as.factor(metadata$CoreID[match(rownames(metaz.ord.b$points),metadata$Sample)])))

plot(metaz.ord.b$points[,1],metaz.ord.b$points[,2],pch=16,col=as.numeric(as.factor(metadata$CoreID[match(rownames(metaz.ord.b$points),metadata$Sample)])),main="Bray 50")
#ordisurf(metaz.ord,metadata$age[match(rownames(metaz.ord$points),metadata$Sample)],add=TRUE)
text(metaz.ord.b$points[,1],metaz.ord.b$points[,2]+0.05,labels=metadata$age[match(rownames(metaz.ord.b$points),metadata$Sample)],cex=0.8)
legend("topleft",bty='n', col=c(1,2,3),pch=16,legend=levels(as.factor(metadata$CoreID[match(rownames(metaz.ord.b$points),metadata$Sample)])))
```

## Taxonomic Overview
Here we are showing taxonomic diversity for all samples across all cores, including only spp with high certainty (>100 reads), showing only 240 out of 322.

```{r, fig.width=12,fig.height=7.5}

par(mfrow = c(1, 1)) 


raw.metazoa.wide.d <- as.data.frame(raw.metazoa.wide)
rownames(raw.metazoa.wide.d) <- raw.metazoa.wide.d[,1]
raw.metazoa.wide.d <- raw.metazoa.wide.d[,-1]
raw.metazoa.wide.d <- raw.metazoa.wide.d[order(rowSums(raw.metazoa.wide.d),decreasing = TRUE),]

strat.plot(t(raw.metazoa.wide.d)[,1:40],yvar = metadata$age[match(colnames(raw.metazoa.wide.d),metadata$Sample)],y.rev=TRUE,plot.line=FALSE,lwd.bar=3,ylabel = "Age(CalYr BP)",cex.ylabel = 0.8,cex.xlabel = 1,xLeft = 0.1,yBottom = 0.1,ylabPos=3,las=2,col.bar = "orange")
strat.plot(t(raw.metazoa.wide.d)[,41:80],yvar = metadata$age[match(colnames(raw.metazoa.wide.d),metadata$Sample)],y.rev=TRUE,plot.line=FALSE,lwd.bar=3,ylabel = "Age(CalYr BP)",cex.ylabel = 0.8,cex.xlabel = 1,xLeft = 0.1,yBottom = 0.1,ylabPos=3,las=2,col.bar = "orange")
strat.plot(t(raw.metazoa.wide.d)[,81:120],yvar = metadata$age[match(colnames(raw.metazoa.wide.d),metadata$Sample)],y.rev=TRUE,plot.line=FALSE,lwd.bar=3,ylabel = "Age(CalYr BP)",cex.ylabel = 0.8,cex.xlabel = 1,xLeft = 0.1,yBottom = 0.1,ylabPos=3,las=2,col.bar = "orange")
strat.plot(t(raw.metazoa.wide.d)[,121:160],yvar = metadata$age[match(colnames(raw.metazoa.wide.d),metadata$Sample)],y.rev=TRUE,plot.line=FALSE,lwd.bar=3,ylabel = "Age(CalYr BP)",cex.ylabel = 0.8,cex.xlabel = 1,xLeft = 0.1,yBottom = 0.1,ylabPos=3,las=2,col.bar = "orange")
strat.plot(t(raw.metazoa.wide.d)[,161:200],yvar = metadata$age[match(colnames(raw.metazoa.wide.d),metadata$Sample)],y.rev=TRUE,plot.line=FALSE,lwd.bar=3,ylabel = "Age(CalYr BP)",cex.ylabel = 0.8,cex.xlabel = 1,xLeft = 0.1,yBottom = 0.1,ylabPos=3,las=2,col.bar = "orange")
strat.plot(t(raw.metazoa.wide.d)[,201:240],yvar = metadata$age[match(colnames(raw.metazoa.wide.d),metadata$Sample)],y.rev=TRUE,plot.line=FALSE,lwd.bar=3,ylabel = "Age(CalYr BP)",cex.ylabel = 0.8,cex.xlabel = 1,xLeft = 0.1,yBottom = 0.1,ylabPos=3,las=2,col.bar = "orange")

```

Let's have a look at the plants, showing all 166.


```{r, fig.width=12,fig.height=7.5}

raw.plants.wide.d <- as.data.frame(raw.plants.wide)
rownames(raw.plants.wide.d) <- raw.plants.wide.d[,1]
raw.plants.wide.d <- raw.plants.wide.d[,-1]
raw.plants.wide.d <- raw.plants.wide.d[order(rowSums(raw.plants.wide.d),decreasing = TRUE),]

strat.plot(t(raw.plants.wide.d)[,1:40],yvar = metadata$age[match(colnames(raw.plants.wide.d),metadata$Sample)],y.rev=TRUE,plot.line=FALSE,lwd.bar=3,ylabel = "Age(CalYr BP)",cex.ylabel = 0.8,cex.xlabel = 1,xLeft = 0.1,yBottom = 0.1,ylabPos=3,las=2,col.bar = "orange")
strat.plot(t(raw.plants.wide.d)[,41:80],yvar = metadata$age[match(colnames(raw.plants.wide.d),metadata$Sample)],y.rev=TRUE,plot.line=FALSE,lwd.bar=3,ylabel = "Age(CalYr BP)",cex.ylabel = 0.8,cex.xlabel = 1,xLeft = 0.1,yBottom = 0.1,ylabPos=3,las=2,col.bar = "orange")
strat.plot(t(raw.plants.wide.d)[,81:120],yvar = metadata$age[match(colnames(raw.plants.wide.d),metadata$Sample)],y.rev=TRUE,plot.line=FALSE,lwd.bar=3,ylabel = "Age(CalYr BP)",cex.ylabel = 0.8,cex.xlabel = 1,xLeft = 0.1,yBottom = 0.1,ylabPos=3,las=2,col.bar = "orange")
strat.plot(t(raw.plants.wide.d)[,121:166],yvar = metadata$age[match(colnames(raw.plants.wide.d),metadata$Sample)],y.rev=TRUE,plot.line=FALSE,lwd.bar=3,ylabel = "Age(CalYr BP)",cex.ylabel = 0.8,cex.xlabel = 1,xLeft = 0.1,yBottom = 0.1,ylabPos=3,las=2,col.bar = "orange")

```

Finally the bacteria, showing only 360 out of 1049.

```{r, fig.width=12,fig.height=7.5}

raw.bacteria.wide.d <- as.data.frame(raw.bacteria.wide)
rownames(raw.bacteria.wide.d) <- raw.bacteria.wide.d[,1]
raw.bacteria.wide.d <- raw.bacteria.wide.d[,-1]
raw.bacteria.wide.d <- raw.bacteria.wide.d[order(rowSums(raw.bacteria.wide.d),decreasing = TRUE),]

strat.plot(t(raw.bacteria.wide.d)[,1:40],yvar = metadata$age[match(colnames(raw.bacteria.wide.d),metadata$Sample)],y.rev=TRUE,plot.line=FALSE,lwd.bar=3,ylabel = "Age(CalYr BP)",cex.ylabel = 0.8,cex.xlabel = 1,xLeft = 0.1,yBottom = 0.1,ylabPos=3,las=2,col.bar = "orange")
strat.plot(t(raw.bacteria.wide.d)[,41:80],yvar = metadata$age[match(colnames(raw.bacteria.wide.d),metadata$Sample)],y.rev=TRUE,plot.line=FALSE,lwd.bar=3,ylabel = "Age(CalYr BP)",cex.ylabel = 0.8,cex.xlabel = 1,xLeft = 0.1,yBottom = 0.1,ylabPos=3,las=2,col.bar = "orange")
strat.plot(t(raw.bacteria.wide.d)[,81:120],yvar = metadata$age[match(colnames(raw.bacteria.wide.d),metadata$Sample)],y.rev=TRUE,plot.line=FALSE,lwd.bar=3,ylabel = "Age(CalYr BP)",cex.ylabel = 0.8,cex.xlabel = 1,xLeft = 0.1,yBottom = 0.1,ylabPos=3,las=2,col.bar = "orange")
strat.plot(t(raw.bacteria.wide.d)[,121:160],yvar = metadata$age[match(colnames(raw.bacteria.wide.d),metadata$Sample)],y.rev=TRUE,plot.line=FALSE,lwd.bar=3,ylabel = "Age(CalYr BP)",cex.ylabel = 0.8,cex.xlabel = 1,xLeft = 0.1,yBottom = 0.1,ylabPos=3,las=2,col.bar = "orange")
strat.plot(t(raw.bacteria.wide.d)[,161:200],yvar = metadata$age[match(colnames(raw.bacteria.wide.d),metadata$Sample)],y.rev=TRUE,plot.line=FALSE,lwd.bar=3,ylabel = "Age(CalYr BP)",cex.ylabel = 0.8,cex.xlabel = 1,xLeft = 0.1,yBottom = 0.1,ylabPos=3,las=2,col.bar = "orange")
strat.plot(t(raw.bacteria.wide.d)[,201:240],yvar = metadata$age[match(colnames(raw.bacteria.wide.d),metadata$Sample)],y.rev=TRUE,plot.line=FALSE,lwd.bar=3,ylabel = "Age(CalYr BP)",cex.ylabel = 0.8,cex.xlabel = 1,xLeft = 0.1,yBottom = 0.1,ylabPos=3,las=2,col.bar = "orange")
strat.plot(t(raw.bacteria.wide.d)[,241:280],yvar = metadata$age[match(colnames(raw.bacteria.wide.d),metadata$Sample)],y.rev=TRUE,plot.line=FALSE,lwd.bar=3,ylabel = "Age(CalYr BP)",cex.ylabel = 0.8,cex.xlabel = 1,xLeft = 0.1,yBottom = 0.1,ylabPos=3,las=2,col.bar = "orange")
strat.plot(t(raw.bacteria.wide.d)[,281:320],yvar = metadata$age[match(colnames(raw.bacteria.wide.d),metadata$Sample)],y.rev=TRUE,plot.line=FALSE,lwd.bar=3,ylabel = "Age(CalYr BP)",cex.ylabel = 0.8,cex.xlabel = 1,xLeft = 0.1,yBottom = 0.1,ylabPos=3,las=2,col.bar = "orange")
strat.plot(t(raw.bacteria.wide.d)[,321:360],yvar = metadata$age[match(colnames(raw.bacteria.wide.d),metadata$Sample)],y.rev=TRUE,plot.line=FALSE,lwd.bar=3,ylabel = "Age(CalYr BP)",cex.ylabel = 0.8,cex.xlabel = 1,xLeft = 0.1,yBottom = 0.1,ylabPos=3,las=2,col.bar = "orange")


```


